File: app/src/main/java/org/tasks/activities/CreateCalendarViewModel.java
package org.tasks.activities;

import org.tasks.caldav.CaldavClient;
import org.tasks.data.CaldavAccount;
import org.tasks.ui.CompletableViewModel;

public class CreateCalendarViewModel extends CompletableViewModel<String> {
  public void createCalendar(CaldavClient client, CaldavAccount account, String name) {
    run(() -> client.forAccount(account).makeCollection(name));
  }
}

File: app/src/main/java/org/tasks/caldav/CreateCalendarViewModel.kt
package org.tasks.caldav

import dagger.hilt.android.lifecycle.HiltViewModel
import org.tasks.data.CaldavAccount
import org.tasks.ui.CompletableViewModel
import javax.inject.Inject

@HiltViewModel
class CreateCalendarViewModel @Inject constructor(
        private val provider: CaldavClientProvider
): CompletableViewModel<String?>() {
    suspend fun createCalendar(account: CaldavAccount, name: String, color: Int) {
        run { provider.forAccount(account).makeCollection(name, color) }
    }
}